config.go

```diff
@@ -1,121 +1,101 @@
-}
-
-// Config holds parameters for config.
+// Config represents the config.
 type Config struct {
-	Path       string
-	TrimSuffix string
-	FileSystem afero.Fs
+	path string
+	data map[string]interface{}
 }
 
-// NewConfig returns a new Config struct with a given set of parameters.
-func NewConfig(path string) *Config {
-	fs := afero.NewOsFs()
-	return &Config{
-		Path:       path,
-		TrimSuffix: DefaultTrimSuffix,
-		FileSystem: fs,
+// Load loads Config from file.
+func Load(options ...func(config *Config)) (*Config, error) {
+	config := &Config{
+		path: DefaultPath,
+	}
+	for _, option := range options {
+		option(config)
 	}
-}
 
-// GetWithFallback returns the configured value of a given key, and the fallback
-// value if no key does not exist.
-func (c *Config) GetWithFallback(key, fallback string) string {
-	value, err := c.Get(key)
-	if err != nil {
-		return fallback
+	if _, err := os.Stat(config.path); os.IsNotExist(err) {
+		return config, errors.Wrap(err, "config file does not exist")
 	}
 
-	return value
-}
+	data, err := ioutil.ReadFile(config.path)
+	if err != nil {
+		return config, errors.Wrap(err, "Failed to read config file")
+	}
 
-// Get returns the configured value of a given key.
-func (c *Config) Get(key string) (string, error) {
-	var value string
+	var configData map[string]interface{}
 
-	// These directories are weighted in order of precedence. Values specified
-	// in multiple paths will be overridden by directories further the list.
-	//
-	// @see https://github.com/skpr/docs/blob/master/docs/config.md#deep-dive
-	paths := []string{
-		c.filePath(PathPartConfig, PathPartDefault, key),
-		c.filePath(PathPartSecret, PathPartDefault, key),
-		c.filePath(PathPartConfig, PathPartOverride, key),
-		c.filePath(PathPartSecret, PathPartOverride, key),
+	err = json.Unmarshal(data, &configData)
+	if err != nil {
+		return config, errors.Wrap(err, "Failed to unmarshal config")
 	}
+	config.data = configData
+
+	return config, nil
+}
 
-	configNoExist := true
-	for _, file := range paths {
-		if _, err := c.FileSystem.Stat(file); os.IsNotExist(err) {
-			continue
-		}
-		contents, err := c.readFile(file)
-		if err != nil {
-			// This could indicate permissions issues with the mount, so exit.
-			return "", err
-		}
-		configNoExist = false
-		value = string(contents)
+// Get returns a string value for the key.
+func (c *Config) Get(key string) (string, bool) {
+	value, ok := c.getValue(key)
+	if value == nil {
+		value = ""
 	}
+	return value.(string), ok
+}
 
-	if configNoExist {
-		err := fmt.Errorf("key not found")
-		return "", err
+// GetBool returns a boolean value for the key.
+func (c *Config) GetBool(key string) (bool, bool) {
+	value, ok := c.getValue(key)
+	if value == nil {
+		value = false
 	}
+	return value.(bool), ok
+}
 
-	return strings.TrimSuffix(value, DefaultTrimSuffix), nil
+// GetInt returns a int value for the key.
+func (c *Config) GetInt(key string) (int, bool) {
```

Comment:
Maybe we could make these return an `error` instead. That way we could:
* Use https://golang.org/pkg/strconv/#Atoi
* Have an error of type "Not found" which we could check for.

Reply:
Discussed with @nickschuch and agreed it's ok to return the ok as a boolean.